File: company-brain-ai/tests/fixtures/python_sqlalchemy/app/repository.py
"""Repository layer reading users.email."""
from sqlalchemy.orm import Session
from .models import User


class UserRepository:
    def __init__(self, db: Session):
        self.db = db

    def find_by_id(self, user_id: int):
        return self.db.query(User).filter(User.id == user_id).first()

    def find_all(self):
        return self.db.query(User.email, User.name).all()

    def find_by_email(self, email: str):
        return self.db.query(User).filter(User.email == email).first()

File: company-brain-ai/tests/fixtures/typescript_drizzle/src/repository.ts
// Repository reading users.email via Drizzle
import { db } from "./db";
import { users } from "./db/schema";
import { eq } from "drizzle-orm";

export class UserRepository {
  async findById(id: number) {
    return db.select().from(users).where(eq(users.id, id)).limit(1);
  }

  async findAll() {
    return db.select({ email: users.email, name: users.name }).from(users);
  }

  async findByEmail(email: string) {
    return db.select().from(users).where(eq(users.email, email)).limit(1);
  }
}
